internal/mode/static/state/store.go

```diff
@@ -17,9 +20,59 @@ type Updater interface {
 
 // objectStore is a store of client.Object
 type objectStore interface {
-	get(nsname types.NamespacedName) client.Object
+	get(objType client.Object, nsname types.NamespacedName) client.Object
 	upsert(obj client.Object)
-	delete(nsname types.NamespacedName)
+	delete(objType client.Object, nsname types.NamespacedName)
+}
+
+// ngfPolicyObjectStore is a store of policies.Policy.
+// A single store should be used to store all types of policies.Policy.
+type ngfPolicyObjectStore struct {
```

Comment:
The store is already an interface, so I think it makes sense to add a new implementation rather than modify the existing one.

Additionally, this store is shared across all ngf policies, whereas the other stores only store one object.